src/test/java/com/rackspace/salus/monitor_management/services/MonitorManagementTest.java

```diff
@@ -1782,13 +1798,46 @@ public void testGetMonitorsFromLabels() {
     // Create monitors which do have the labels we care about
     createMonitorsForTenant(monitorsWithLabels, tenantId, labels);
 
+    // Create a "select all" type of monitor where label selector is empty
+    createMonitorsForTenant(1, tenantId, Collections.emptyMap());
+
     entityManager.flush();
 
     Page<Monitor> monitors = monitorManagement.getMonitorsFromLabels(labels, tenantId, Pageable.unpaged());
-    assertEquals(monitorsWithLabels, monitors.getTotalElements());
+    assertEquals(monitorsWithLabels+1, monitors.getTotalElements());
     assertNotNull(monitors);
   }
 
+  @Test
+  public void testGetMonitorsFromLabelsOnlyReturnsTenantMonitors() {
+    int monitorsWithLabels = new Random().nextInt(10) + 10;
+    int monitorsWithoutLabels = new Random().nextInt(10) + 20;
+    int monitorsThatMatchEverything = new Random().nextInt(10) + 20;
+    int monitorsWithResourceId = new Random().nextInt(10) + 20;
+    String tenantId = RandomStringUtils.randomAlphabetic(10);
+
+    Map<String, String> labels = Collections.singletonMap("mykey", "myvalue");
+
+    // Create monitors which don't have the labels we care about
+    createMonitorsForTenant(monitorsWithoutLabels, tenantId);
```

Comment:
```suggestion
    createMonitorsForTenant(monitorsWithoutLabels, tenantId, Map.of("label", "no-match");
```

Would something like this make more sense?  Currently this seems to be leaving `labelSelector` as null or empty?  and that makes me wonder why the test is passing.

Reply:
From debugging through this I can confirm that the testing harness is actually adding random values for the labels here. I can certainly make it clearer by doing what you suggest and will update the `monitorsWithoutLabels` to something more relevant. Or I could just update `monitorsWithoutLabels` and leave it at that.

Reply:
I'm guessing it's matching because the podam is creating the dummy labels that don't match.  If that is the case maybe it is sufficient to the name of the count from "monitorsWithoutLabels"  to "monitorsWithNonMatchingLabels"?

Reply:
oops, didn't see @harageth's comment before i wrote mine.  sounds like we are basically saying the same thing.

Reply:
Oh right, I overlooked that podam was involved here.

Reply:
and yeah, a name change could be helpful